app_backend/git_client.py

```diff
@@ -0,0 +1,162 @@
+    except FileNotFoundError as e:
+        raise GitError("Git is not installed!") from e
+
+
+def run_gh(*args: str) -> str:
+    """Run a GitHub CLI command and return its output."""
+    try:
+        result = subprocess.run(["gh", *args], capture_output=True, text=True, check=True)
+        return result.stdout.strip()
+    except subprocess.CalledProcessError as e:
+        raise GitError(f"GitHub CLI command failed ({e.returncode}): {e.stderr.strip()}") from e
+    except FileNotFoundError as e:
+        raise GitError("GitHub CLI (gh) is not installed!") from e
+
+
+class GitCommit:
+    """Represents a git commit with relevant information."""
+
+    def __init__(self, hash: str, message: str, author: str, date: str):
+        self.hash = hash
+        self.message = message
+        self.author = author
+        self.date = date
+
+    def __repr__(self) -> str:
+        return f"GitCommit(hash='{self.hash[:8]}', message='{self.message[:50]}...')"
+
+
+class GitClient:
+    """Client for git operations, mimicking the existing metta git utilities."""
+
+    DEFAULT_BRANCH = "main"
+
+    def get_current_commit(self) -> str:
+        """Get the current git commit hash."""
+        return run_git("rev-parse", "HEAD")
+
+    def get_current_branch(self) -> str:
+        """Get the current git branch name."""
+        try:
+            return run_git("symbolic-ref", "--short", "HEAD")
+        except GitError as e:
+            if "not a git repository" in str(e):
+                raise ValueError("Not in a git repository") from e
+            elif "HEAD is not a symbolic ref" in str(e):
+                return self.get_current_commit()
+            raise
+
+    def get_commit_message(self, commit_hash: str) -> str:
+        """Get the commit message for a given commit."""
+        return run_git("log", "-1", "--pretty=%B", commit_hash)
+
+    def validate_git_ref(self, ref: str) -> Optional[str]:
+        """Validate a git reference exists (locally or in remote)."""
+        try:
+            commit_hash = run_git("rev-parse", "--verify", ref)
+            return commit_hash
+        except GitError:
+            return None
+
+    def commit_exists(self, commit_hash: str) -> bool:
+        """Check if commit exists locally."""
+        try:
+            run_git("cat-file", "-e", commit_hash)
+            return True
+        except GitError:
+            return False
+
+    def get_merge_base(self, commit_hash: str, base_branch: str = DEFAULT_BRANCH) -> str:
+        """Get merge base between commit and base branch."""
+        try:
+            return run_git("merge-base", base_branch, commit_hash)
+        except GitError:
+            # Fallback to base branch
+            logger.warning(f"Could not find merge base for {commit_hash}, using {base_branch}")
+            return base_branch
+
+    def get_commit_range(self, commit_hash: str, base_branch: str = DEFAULT_BRANCH) -> List[GitCommit]:
```

Comment:
What if commit_hash is equal to base_branch? Make sure this case is covered in code and in tests